### PR TITLE
Feature/main cmdline app

### PR DESCRIPTION
**Pull request by dchvs99**

### CODE REVIEW

File: rr/ai/ai_manager_mock.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+        img = ImageHandler.buffer_to_np_array(
+            image.get_data(), image.get_width(), image.get_height())
+
+        image_preprocessed = self.preprocess_detection(img)
+
+#        inference_results = self.run_inference(image_preprocessed)
+
+#        image_postprocessed = self.postprocess_detection(
+#            img, inference_results)
```
Comment (emmadrigal) [inline rr/ai/ai_manager_mock.py]: Don't commit dirty code

File: rr/ai/ai_manager_mock.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        size = h * w * c
+        buffer = GstUtils.buffer_new_wrapped_full(
+            image_postprocessed.tobytes(), size)
+
+        # Create GstImage
+        sample = image.get_sample()
+        caps = sample.get_caps()
+        sample2 = GstUtils.sample_new(buffer, caps)
+        fmt = 'RGB'
+        image2 = GstImage(w, h, fmt, sample2, image.get_media())
```
Comment (emmadrigal) [inline rr/ai/ai_manager_mock.py]: Are you using `image2`?

File: rr/gstreamer/gst_media.py [hunk 3]
```diff
@@ -146,6 +150,34 @@ def get_media(self):
...
+    def make(cls, desc, all_triggers):
+        pipe = 'uridecodebin uri=%s caps=video/x-h264 ! queue ! h264parse ! avdec_h264 ! queue ! videoconvert ! videoscale ! video/x-raw,width=320,height=240,format=RGB ! appsink emit-signals=true name=appsink' % (
+            desc["uri"])
+        media = GstMedia()
+        media.create_media(desc['id'], pipe)
+
+        media_triggers = []
+        for trigger in desc['triggers']:
+            match = next(
+                (candidate for candidate in all_triggers if trigger == candidate._name),
```
Comment (emmadrigal) [inline rr/gstreamer/gst_media.py]: `_name` is supposed to be private

**Review by emmadrigal — CHANGES_REQUESTED** [review]

File: rr/ai/ai_manager_mock.py (hunk 2 above)
Comment (dchvs99) [reply rr/ai/ai_manager_mock.py]: I have deleted this mocked AIManager!

File: rr/ai/ai_manager_mock.py (hunk 1 above)
Comment (dchvs99) [reply rr/ai/ai_manager_mock.py]: Thanks. Also, I have deleted this mocking file as we put the real AIManager to work within this PR.

**Review by emmadrigal — APPROVED** [review]

**Pull request merged**